samples/extensions/sparse_image/sparse_image.cpp

```diff
@@ -0,0 +1,1532 @@
+		virtual_texture.sparse_image_memory_bind[page].memoryOffset = pageOffset;
+	}
+
+	VkBindSparseInfo bind_sparse_info = vkb::initializers::bind_sparse_info();
+	bind_sparse_info.bufferBindCount  = 0U;
+	bind_sparse_info.pBufferBinds     = nullptr;
+
+	VkSparseImageMemoryBindInfo sparse_image_memory_bind_info{};
+	sparse_image_memory_bind_info.image     = virtual_texture.texture_image;
+	sparse_image_memory_bind_info.bindCount = virtual_texture.sparse_image_memory_bind.size();
+	sparse_image_memory_bind_info.pBinds    = virtual_texture.sparse_image_memory_bind.data();
+
+	bind_sparse_info.imageBindCount = 1U;
+	bind_sparse_info.pImageBinds    = &sparse_image_memory_bind_info;
+
+	VkSparseImageOpaqueMemoryBindInfo sparse_image_opaque_memory_bind_info{};
+	sparse_image_opaque_memory_bind_info.image     = nullptr;
+	sparse_image_opaque_memory_bind_info.bindCount = 0U;
+	sparse_image_opaque_memory_bind_info.pBinds    = nullptr;
+
+	bind_sparse_info.imageOpaqueBindCount = 0U;
+	bind_sparse_info.pImageOpaqueBinds    = nullptr;
+
+	bind_sparse_info.signalSemaphoreCount = 0U;
+	bind_sparse_info.pSignalSemaphores    = nullptr;
+	bind_sparse_info.waitSemaphoreCount   = 0U;
+	bind_sparse_info.pWaitSemaphores      = nullptr;
+
+	VkFence           fence;
+	VkFenceCreateInfo fence_info{};
+	fence_info.sType = VK_STRUCTURE_TYPE_FENCE_CREATE_INFO;
+	fence_info.flags = VK_FLAGS_NONE;
+
+	VK_CHECK(vkCreateFence(get_device().get_handle(), &fence_info, nullptr, &fence));
+	vkQueueBindSparse(sparse_queue, 1U, &bind_sparse_info, fence);
+	VK_CHECK(vkWaitForFences(get_device().get_handle(), 1, &fence, VK_TRUE, DEFAULT_FENCE_TIMEOUT));
+	vkDestroyFence(get_device().get_handle(), fence, nullptr);
+}
+
+/**
+ * 	@brief Return information on what mip_level the particular page_index belongs to.
+ */
+uint8_t SparseImage::get_mip_level(size_t page_index)
+{
+	uint8_t mip_level = 0xFF;
+	if (virtual_texture.mip_levels == 1U)
+	{
+		return virtual_texture.base_mip_level;
+	}
+
+	for (uint32_t i = virtual_texture.base_mip_level; i < virtual_texture.mip_levels; i++)
+	{
+		if (page_index < (virtual_texture.mip_properties[i].mip_base_page_index + virtual_texture.mip_properties[i].mip_num_pages))
+		{
+			mip_level = i;
+			break;
+		}
+	}
+	return mip_level;
+}
+
+/**
+ * 	@brief Calculate dependencies, fill the required lists and set required flags for the particular BLOCK to be updated on screen.
+ */
+void SparseImage::process_texture_block(const TextureBlock &texture_block)
+{
+	std::list<size_t> memory_index_list;
+
+	if (texture_block.old_mip_level < virtual_texture.mip_levels)        // TODO(GS): This condition seems to be completely redundant but for some reason app is crashing with vector subscript being out of range.
+	{
+		// Old value calculations and removal from the render required list
+		get_memory_dependency_for_the_block(texture_block.column, texture_block.row, texture_block.old_mip_level, memory_index_list);
```

Comment:
Does this function add values to a potentially non-empty `memory_index_list`?
If instead it just determines some indices, it might be better if it would not take a `some_container<size_t> &` but returns a `some_container<size_t>`. And I would use a `std::vector`, instead of a `std::list`.